# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1232452724 18000
#      Tue Jan 20 06:58:44 2009 -0500
# Node ID f4a4f914462ed74bd125b28ccaaf7a76c09dc762
# Parent  17ef48c699a508ff443e95de5b66389a6aaf6f7b
assert.m: use isfield instead of struct_contains

diff --git a/scripts/ChangeLog b/scripts/ChangeLog
--- a/scripts/ChangeLog
+++ b/scripts/ChangeLog
@@ -1,8 +1,12 @@
+2009-01-20  John W. Eaton  <jwe@octave.org>
+
+	* testfun/assert.m: Use isfield instead of struct_contains.
+
 2009-01-17 Ben Abbott <bpabbott@mac.com>
 
 	* general/cart2sph.m, cart2pol.m, sph2cart.m pol2cart.m:
 	Permit scalars when transforming coordinates.
 
 2009-01-17  Steven Verstoep <isgoed@hotmail.com>
 
 	* plot/__quiver__.m: __quiver__.m: Fix auto-size of (x,y) coord.
diff --git a/scripts/testfun/assert.m b/scripts/testfun/assert.m
--- a/scripts/testfun/assert.m
+++ b/scripts/testfun/assert.m
@@ -113,17 +113,17 @@ function assert (cond, varargin)
       if (! isstruct (cond) || any (size (cond) != size (expected))
 	  || rows (fieldnames (cond)) != rows (fieldnames (expected)))
 	iserror = 1;
       else
 	try
 	  empty = numel (cond) == 0;
 	  normal = numel (cond) == 1;
 	  for [v, k] = cond
-	    if (! struct_contains (expected, k))
+	    if (! isfield (expected, k))
 	      error ();
 	    endif
 	    if (empty)
 	      v = cell (1, 0);
 	    endif
 	    if (normal)
 	      v = {v};
 	    else
